kconfiglib.py

```diff
@@ -1102,31 +1102,31 @@ def _open(self, filename):
                         "unset" if self.srctree is None else
                         '"{}"'.format(self.srctree)))
 
-    def _enter_file(self, filename, relative_source=False):
+    def _translate_relative_path(self, path):
+        # We expect strictly relative path, not absolute
+        if os.path.isabs(path):
+            raise KconfigSyntaxError(
+                "\n{}:{}: Relative path expected, while '{}' is absolute.\n"
+                "Backtrace:\n{}"
+                    .format(self._filename, self._linenr, path,
+                            "\n".join("{}:{}".format(name, linenr)
+                                      for _, name, linenr
+                                      in reversed(self._filestack))))
```

Comment:
I think it would be better to remove this check and just specify that `rsource` works the same as `source` for absolute paths ("If an absolute path is given to 'source' or 'rsource', it will be opened directly, ignoring `$srctree` and (for 'rsource') the location of the current Kconfig file", or something like that, in the new section that describes `rsource`).

`os.path.join("foo", "/bar")` ignores `"foo"` and returns `"/bar"` btw, so `join()` will "just work" there.

After that, the logic will be so short that `_translate_relative_path()` could be removed.

Reply:
This might be slightly better still:

    Like for 'source', an absolute path given to `rsource` will be opened directly,
    ignoring $srctree and the location of the current Kconfig file.

Reply:
Agree. Absolute paths allowed, `_translate_relative_path()` removed.